mantidproject/mantid
# Workbench plot dialog validation

**Description of work.**
Changes the spectra box in the workbench plot 1D selection dialog to list out the valid options allowing for breaks in the range such as 1,3,5,7-20,500-666.  Also, validation uses the full list too, so it should no longer5 be possible to enter an invalid spectrum, and even if you do manage hen it should present as a log error, not a crash.

**To test:**

1. open workbench
1. load a file like OSI89757, entering a list in the SpectrumList property such as 1,3,5, 7-20
1. Right click the workspace plot->Spectrum ...
1.  The spectrum number box should list the correct range
1, the validation should allow choices within that range, including mixing list and ranges like 1,7,9,11-15,16
1, Load up a second workspace with all the spectra
1. Select both of them with shift & click
1. Right click the workspace plot->Spectrum ...
1.  The spectrum number box should list the range that is available in both (or indeed all if you go for more than 2) workspaces




Fixes #28275



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
